concepts/fundamentals/introduction.md

```diff
@@ -0,0 +1,138 @@
+# Introduction
+
+We'll refer to the _language_ as **AWK** and to the _executable_ that interprets AWK programs as **`awk`** or **`gawk`**.
+
+## Patterns and Actions
+
+An AWK program is structured in **pattern-action pairs**
+
+```awk
+pattern1 { action1 }
+pattern2 { action2 }
+...
+```
+
+A **pattern** can be a regular expression pattern to match against the input.
+But it can also be any expression that results in a true/false value.
+
+An **action** is one or more AWK commands.
+The enclosing braces are required.
+
+If the result of the pattern is _true_, then the corresponding action is evaluated.
+
+### Omit the pattern
+
+The pattern can be omitted, in which case the action is evaluated _for every input record_.
+In other words, an "empty" pattern is implicitly _true_.
+
+### Omit the action
+
+The action can be omitted, in which case the default action is to **print** the current record.
+
+### Special patterns
+
+Two special pattern that are often used are
+
+- `BEGIN` will evaluate the associated action before any input is consumed.
+- `END` will evaluate the associated action after all input is consumed.
+
+## Records and Fields
+
+AWK splits the input into **records** and **fields**.
+The default behaviour is to 
+
+- split the input into records using **newline** as the "record separator", so each _line_ is a record.
+- split the record into fields using **whitespace** as the "field separator", so each _word_ is a field.
+
+You can supply your own input record separator with the `RS` variable.
+You can supply your own input field separator with the `FS` variable.
+Set these variables in the `BEGIN` block.
+
+### Field variables
+
+You refer to fields as `$1`, `$2`, `$3`, etc.
+
+The current record is `$0`.
+
+### Other common variables
+
+- **`NF`** stores the _number of fields_ in the current record.
+- **`$NF`** is the _value of the last field_.
```

Comment:
`$NF` isn't a variable, per se. You may want to be more explicit about `$` proceeding *any* number, ie `$(NF - 1)` is also valid.